server/src/main/java/org/opensearch/rest/action/admin/indices/RestViewAction.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.indices;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.view.CreateViewAction;
+import org.opensearch.action.admin.indices.view.SearchViewAction;
+import org.opensearch.action.search.SearchAction;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.ValidationException;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.NamedRoute;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestCancellableNodeClient;
+import org.opensearch.rest.action.RestStatusToXContentListener;
+import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.rest.action.search.RestSearchAction;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.IntConsumer;
+
+import static org.opensearch.rest.RestRequest.Method.GET;
+import static org.opensearch.rest.RestRequest.Method.POST;
+
+/** All rest handlers for view actions */
+public class RestViewAction {
+
+    private final static Logger LOG = LogManager.getLogger(RestViewAction.class);
+
+    public static final String VIEW_ID = "view_id";
+    public static final String VIEW_ID_PARAMETER = "{" + VIEW_ID + "}";
+
+    /** Handler for create view */
+    public static class CreateViewHandler extends BaseRestHandler {
+
+        @Override
+        public List<Route> routes() {
+            return List.of(new NamedRoute.Builder().path("/views").method(POST).uniqueName(CreateViewAction.NAME).build());
+        }
+
+        @Override
+        public String getName() {
+            return CreateViewAction.NAME;
+        }
+
+        @Override
+        protected RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+            try (final XContentParser parser = request.contentParser()) {
+                final CreateViewAction.Request createViewAction = CreateViewAction.Request.fromXContent(parser);
+                return channel -> client.admin().indices().createView(createViewAction, new RestToXContentListener<>(channel));
+            }
+        }
+    }
+
+    public static class SearchViewHandler extends BaseRestHandler {
+        @Override
+        public List<Route> routes() {
+            return List.of(
+                new NamedRoute.Builder().path("/views/" + VIEW_ID_PARAMETER + "/_search")
+                    .method(GET)
+                    .uniqueName("cluster:views:search")
```

Comment:
should this be: `cluster:admin:views:search` to match with the action name? or better yet, replace it with SearchViewAction.NAME just like in CreateViewHandler

Reply:
Thanks for calling this out; after brainstorming I've landed on the following
- admin actions `cluster:admin/views/create`
- user actions `views:data/read/search`

Let me know what you think

Reply:
I like these. We are introducing a new type of action here, not sure if there are any boxes that need to be checked before introducing a new type of base action. Probly the maintainers can better answer that.